--- SOAPdenovo2-src-r240/standardPregraph/Makefile	2013-07-08 21:00:33.000000000 -0700
+++ patch-files/Makefile.standard	2014-04-01 07:30:02.470481760 -0700
@@ -5,13 +5,13 @@
 #
 ###################################
 
-CC=             gcc
+CC=             icc
 GCCVERSIONMAJOR := $(shell expr `$(CC) -dumpversion | cut -f1 -d.` \>= 4)
 GCCVERSIONMINOR := $(shell expr `$(CC) -dumpversion | cut -f2 -d.` \>= 4)
 ifdef debug
 CFLAGS=         -O0 -g -fomit-frame-pointer #-static #-mcrc32 -march=core2 -msse4.1 -msse4.2
 else
-CFLAGS=         -O4 -fomit-frame-pointer #-static #-mcrc32 -march=core2 -msse4.1 -msse4.2
+CFLAGS=         -O3 -fomit-frame-pointer -xavx
 endif
 DFLAGS=         
 OBJS=		arc.o attachPEinfo.o bubble.o check.o compactEdge.o \
