--- Makefile.sparse	2015-06-19 21:19:30.000000000 -0700
+++ patch-files/Makefile.sparse	2015-06-19 21:20:05.000000000 -0700
@@ -1,4 +1,4 @@
-CC=            g++ # /opt/blc/gcc-4.5.0/bin/gcc #gcc
+CXX=            g++ # /opt/blc/gcc-4.5.0/bin/gcc #gcc
 ifdef debug
 CFLAGS=         -O0 -g -fomit-frame-pointer #-static #-mcrc32 -march=core2 -msse4.1 -msse4.2
 else
@@ -62,7 +62,7 @@
 
 .cpp.o:
 		@printf "Compiling $<...                             \r"; \
-		$(CC) -c $(CFLAGS) $(DFLAGS) $(INCLUDES) $< || echo "Error in command: $(CC) -c $(CFLAGS) $(DFLAGS) $(INCLUDES) $<"
+		$(CXX) -c $(CFLAGS) $(DFLAGS) $(INCLUDES) $< || echo "Error in command: $(CXX) -c $(CFLAGS) $(DFLAGS) $(INCLUDES) $<"
 
 all:		clean $(OBJS) 
 #pregraph_sparse
@@ -74,7 +74,7 @@
 
 pregraph_sparse: clean	envTest $(OBJS)
 		@printf "Linking...                                                       \r"
-		#@$(CC) $(CFLAGS)$(INCLUDES) -o $(PROG)  $(OBJS) $(LIBPATH) $(LIBS) $(ENTRAFLAGS)
+		#@$(CXX) $(CFLAGS)$(INCLUDES) -o $(PROG)  $(OBJS) $(LIBPATH) $(LIBS) $(ENTRAFLAGS)
 		@printf "$(PROG) compilation done.\n";
 
 clean:
