--- test_popgenstat.py	2014-02-25 15:14:41.000000000 -0800
+++ patch-files/test_popgenstat.py	2014-02-25 15:10:33.000000000 -0800
@@ -67,7 +67,7 @@
 
 class SinglePopTest(extendedtest.ExtendedTestCase):
 
-    data = dendropy.DnaCharacterMatrix.get_from_path(pathmap.char_source_path('COII_Apes.nex'), schema="nexus")
+    data = dendropy.DnaCharacterMatrix.get_from_path(pathmap.char_source_path('ROLLPATH/test/COII_Apes.nex'), schema="nexus")
 
     def test_num_segregating_sites(self):
         self.assertEqual(popgenstat.num_segregating_sites(self.data, ignore_uncertain=True), 183)
@@ -87,7 +87,7 @@
 class PopulationPairSummaryStatisticsTests(extendedtest.ExtendedTestCase):
 
     def testPopulationPairSummaryStatistics(self):
-        seqs = dendropy.DnaCharacterMatrix.get_from_path(pathmap.char_source_path('orti.nex'), schema="nexus")
+        seqs = dendropy.DnaCharacterMatrix.get_from_path(pathmap.char_source_path('ROLLPATH/test/orti.nex'), schema="nexus")
         p1 = []
         p2 = []
         for idx, t in enumerate(seqs.taxon_set):
