--- hmmer-3.1b2/testsuite/i10-duplicate-names.pl	2015-02-26 04:46:04.000000000 -0800
+++ patch-files/i10-duplicate-names.pl	2016-05-05 22:37:43.672369146 -0700
@@ -19,12 +19,12 @@
 
 
 # Verify that we have all the executables we need for the test.
-if (! -x "$builddir/src/hmmbuild")   { die "FAIL: didn't find hmmbuild binary in $builddir/src\n";  }
-if (! -x "$builddir/src/hmmpress")   { die "FAIL: didn't find hmmpress binary in $builddir/src\n";  }
-if (! -x "$builddir/src/hmmsearch")  { die "FAIL: didn't find hmmsearch binary in $builddir/src\n"; }
-if (! -x "$builddir/src/hmmscan")    { die "FAIL: didn't find hmmscan binary in $builddir/src\n";   }
-if (! -x "$builddir/src/phmmer")     { die "FAIL: didn't find phmmer binary in $builddir/src\n"; }
-if (! -x "$builddir/src/jackhmmer")  { die "FAIL: didn't find jackhmmer binary in $builddir/src\n";   }
+if (! -x "$builddir/hmmbuild")   { die "FAIL: didn't find hmmbuild binary in $builddir/src\n";  }
+if (! -x "$builddir/hmmpress")   { die "FAIL: didn't find hmmpress binary in $builddir/src\n";  }
+if (! -x "$builddir/hmmsearch")  { die "FAIL: didn't find hmmsearch binary in $builddir/src\n"; }
+if (! -x "$builddir/hmmscan")    { die "FAIL: didn't find hmmscan binary in $builddir/src\n";   }
+if (! -x "$builddir/phmmer")     { die "FAIL: didn't find phmmer binary in $builddir/src\n"; }
+if (! -x "$builddir/jackhmmer")  { die "FAIL: didn't find jackhmmer binary in $builddir/src\n";   }
 
 # Create our test files
 if (! open(ALI1, ">$tmppfx.sto")) { print "FAIL: couldn't open $tmppfx.sto for write";  exit 1; }
@@ -60,15 +60,15 @@
 close SEQ1;
 
 # Build profiles from the test alignments
-@output = `$builddir/src/hmmbuild $tmppfx.hmm $tmppfx.sto 2>&1`;
+@output = `$builddir/hmmbuild $tmppfx.hmm $tmppfx.sto 2>&1`;
 if ($? != 0) { die "FAIL: hmmbuild failed\n"; }
-@output = `$builddir/src/hmmpress $tmppfx.hmm             2>&1`;
+@output = `$builddir/hmmpress $tmppfx.hmm             2>&1`;
 if ($? != 0) { die "FAIL: hmmpress failed\n"; }
 
 
 
 # phmmer should show four results
-$output = `$builddir/src/phmmer --tblout $tmppfx.tbl $tmppfx.fa $tmppfx.fa 2>&1`;
+$output = `$builddir/phmmer --tblout $tmppfx.tbl $tmppfx.fa $tmppfx.fa 2>&1`;
 if ($? != 0) { die "FAIL: phmmer failed\n"; }
 
 &h3::ParseTbl("$tmppfx.tbl");
@@ -76,7 +76,7 @@
 
 
 # jackhmmer should show four results
-$output = `$builddir/src/jackhmmer --tblout $tmppfx.tbl $tmppfx.fa $tmppfx.fa 2>&1`;
+$output = `$builddir/jackhmmer --tblout $tmppfx.tbl $tmppfx.fa $tmppfx.fa 2>&1`;
 if ($? != 0) { die "FAIL: jackhmmer failed\n"; }
 
 &h3::ParseTbl("$tmppfx.tbl");
@@ -84,7 +84,7 @@
 
 
 # hmmsearch should show four results
-$output = `$builddir/src/hmmsearch --tblout $tmppfx.tbl $tmppfx.hmm $tmppfx.fa 2>&1`;
+$output = `$builddir/hmmsearch --tblout $tmppfx.tbl $tmppfx.hmm $tmppfx.fa 2>&1`;
 if ($? != 0) { die "FAIL: hmmsearch failed\n"; }
 
 &h3::ParseTbl("$tmppfx.tbl");
@@ -93,7 +93,7 @@
 
 
 # hmmscan should show four results
-$output = `$builddir/src/hmmscan --tblout $tmppfx.tbl $tmppfx.hmm $tmppfx.fa 2>&1`;
+$output = `$builddir/hmmscan --tblout $tmppfx.tbl $tmppfx.hmm $tmppfx.fa 2>&1`;
 if ($? != 0) { die "FAIL: hmmscan failed\n"; }
 
 &h3::ParseTbl("$tmppfx.tbl");
