--- hmmer-3.1b2/testsuite/i11-hmmalign-mapali.pl	2015-02-26 04:46:05.000000000 -0800
+++ patch-files/i11-hmmalign-mapali.pl	2016-05-05 22:37:43.673369143 -0700
@@ -15,8 +15,8 @@
 $tmppfx     = shift;
 
 # Verify that we have all the executables we need for the test.
-if (! -x "$builddir/src/hmmbuild")   { die "FAIL: didn't find hmmbuild binary in $builddir/src\n";  }
-if (! -x "$builddir/src/hmmalign")   { die "FAIL: didn't find hmmalign binary in $builddir/src\n";  }
+if (! -x "$builddir/hmmbuild")   { die "FAIL: didn't find hmmbuild binary in $builddir/src\n";  }
+if (! -x "$builddir/hmmalign")   { die "FAIL: didn't find hmmalign binary in $builddir/src\n";  }
 
 
 # Create our test files.
@@ -49,7 +49,7 @@
 close ALI1;
 close SEQ2;
 
-$output = `$builddir/src/hmmbuild -O $tmppfx.sto2 $tmppfx.hmm $tmppfx.sto 2>&1`;
+$output = `$builddir/hmmbuild -O $tmppfx.sto2 $tmppfx.hmm $tmppfx.sto 2>&1`;
 if ($? != 0) { die "FAIL: hmmbuild failed\n"; }
 
 ($resave) = ($output =~ /processed alignment resaved to:\s+(\S+)/);
@@ -65,7 +65,7 @@
 if ($aseq7 ne "ACDEFAHIKLMNPQRSTVWY") { die "FAIL: D->I transition wasn't trace doctored properly?\n"; }
 if ($aseq8 ne "ACDEFGHIKAMNPQRSTVWY") { die "FAIL: I->D transition wasn't trace doctored properly?\n"; }
 
-$output = `$builddir/src/hmmalign --mapali $tmppfx.sto $tmppfx.hmm $tmppfx.fa 2>&1`;
+$output = `$builddir/hmmalign --mapali $tmppfx.sto $tmppfx.hmm $tmppfx.fa 2>&1`;
 if ($? != 0) { die "FAIL: hmmalign --mapali failed\n"; }
 
 ($testseq) = ($output =~ /\ntest\s+(\S+)/);
