--- hmmer-3.2.1/testsuite/i12-delete-corruption.pl	2018-06-13 07:39:03.000000000 -0700
+++ patch-files/i12-delete-corruption.pl	2018-10-24 17:04:13.895513077 -0700
@@ -20,8 +20,8 @@
 }
 
 # Verify that we have all the executables we need for the test.
-if (! -x "$builddir/src/hmmbuild")   { die "FAIL: didn't find hmmbuild binary in $builddir/src\n";  }
-if (! -x "$builddir/src/hmmalign")   { die "FAIL: didn't find hmmalign binary in $builddir/src\n";  }
+if (! -x "../bin/hmmbuild")   { die "FAIL: didn't find hmmbuild binary in $builddir\n";  }
+if (! -x "../bin/hmmalign")   { die "FAIL: didn't find hmmalign binary in $builddir\n";  }
 
 # Create our test files.
 # The test seqs are missing one column, K, in an unambiguous small alignment.
@@ -46,16 +46,16 @@
 close ALI1;
 close SEQ1;
 
-@output = `$builddir/src/hmmbuild $tmppfx.hmm $tmppfx.sto 2>&1`;
+@output = `hmmbuild $tmppfx.hmm $tmppfx.sto 2>&1`;
 if ($? != 0) { die "FAIL: hmmbuild failed\n"; }
 
-$output = `$builddir/src/hmmalign $tmppfx.hmm $tmppfx.fa 2>&1`;
+$output = `hmmalign $tmppfx.hmm $tmppfx.fa 2>&1`;
 if ($? != 0) { die "FAIL: hmmalign failed\n"; }
 
 ($testseq) = ($output =~ /\ntest1\s+(\S+)/);
 if ($testseq ne "ACDEFGHI-LMNPQRSTVWY") { die "FAIL: test sequence corrupted by hmmalign\n"; }
 
-#$output = `$builddir/src/hmmalign --allcol $tmppfx.hmm $tmppfx.fa 2>&1`;
+#$output = `$builddir/hmmalign --allcol $tmppfx.hmm $tmppfx.fa 2>&1`;
 #if ($? != 0) { die "FAIL: hmmalign failed\n"; }
 #
 #($testseq) = ($output =~ /\ntest1\s+(\S+)/);
