--- hmmer-3.1b2/testsuite/i12-delete-corruption.pl	2015-02-26 04:46:05.000000000 -0800
+++ patch-files/i12-delete-corruption.pl	2016-05-05 22:37:43.673369143 -0700
@@ -21,8 +21,8 @@
 }
 
 # Verify that we have all the executables we need for the test.
-if (! -x "$builddir/src/hmmbuild")   { die "FAIL: didn't find hmmbuild binary in $builddir/src\n";  }
-if (! -x "$builddir/src/hmmalign")   { die "FAIL: didn't find hmmalign binary in $builddir/src\n";  }
+if (! -x "$builddir/hmmbuild")   { die "FAIL: didn't find hmmbuild binary in $builddir/src\n";  }
+if (! -x "$builddir/hmmalign")   { die "FAIL: didn't find hmmalign binary in $builddir/src\n";  }
 
 # Create our test files.
 # The test seqs are missing one column, K, in an unambiguous small alignment.
@@ -47,16 +47,16 @@
 close ALI1;
 close SEQ1;
 
-@output = `$builddir/src/hmmbuild $tmppfx.hmm $tmppfx.sto 2>&1`;
+@output = `$builddir/hmmbuild $tmppfx.hmm $tmppfx.sto 2>&1`;
 if ($? != 0) { die "FAIL: hmmbuild failed\n"; }
 
-$output = `$builddir/src/hmmalign $tmppfx.hmm $tmppfx.fa 2>&1`;
+$output = `$builddir/hmmalign $tmppfx.hmm $tmppfx.fa 2>&1`;
 if ($? != 0) { die "FAIL: hmmalign failed\n"; }
 
 ($testseq) = ($output =~ /\ntest1\s+(\S+)/);
 if ($testseq ne "ACDEFGHI-LMNPQRSTVWY") { die "FAIL: test sequence corrupted by hmmalign\n"; }
 
-#$output = `$builddir/src/hmmalign --allcol $tmppfx.hmm $tmppfx.fa 2>&1`;
+#$output = `$builddir/hmmalign --allcol $tmppfx.hmm $tmppfx.fa 2>&1`;
 #if ($? != 0) { die "FAIL: hmmalign failed\n"; }
 #
 #($testseq) = ($output =~ /\ntest1\s+(\S+)/);
