--- hmmer-3.1b2/testsuite/i15-hmmconvert.pl	2015-02-26 04:46:05.000000000 -0800
+++ patch-files/i15-hmmconvert.pl	2016-05-05 22:37:43.673369143 -0700
@@ -15,16 +15,16 @@
 }
 
 # Verify that we have all the executables we need for the test.
-if (! -x "$builddir/src/hmmconvert") { die "FAIL: didn't find hmmconvert binary in $builddir/src\n";  }
+if (! -x "$builddir/hmmconvert") { die "FAIL: didn't find hmmconvert binary in $builddir/\n";  }
 
 # Create our test files (data embedded in script as functions: see end of file
 &h2_nt_file("$tmppfx.nt.hmm");
 &h2_aa_file("$tmppfx.aa.hmm");
 
-@output = `$builddir/src/hmmconvert $tmppfx.nt.hmm 2>&1`;
+@output = `$builddir/hmmconvert $tmppfx.nt.hmm 2>&1`;
 if ($? != 0) { die "FAIL: hmmconvert of H2 nucleic model failed\n"; }
 
-@output = `$builddir/src/hmmconvert $tmppfx.aa.hmm 2>&1`;
+@output = `$builddir/hmmconvert $tmppfx.aa.hmm 2>&1`;
 if ($? != 0) { die "FAIL: hmmconvert of H2 protein model failed\n"; }
 
 
