--- hmmer-3.2.1/testsuite/i16-build-allins.pl	2018-06-13 07:39:03.000000000 -0700
+++ patch-files/i16-build-allins.pl	2018-10-23 16:16:33.265979700 -0700
@@ -15,7 +15,7 @@
 }
 
 # Verify that we have all the executables we need for the test.
-if (! -x "$builddir/src/hmmbuild") { die "FAIL: didn't find hmmconvert binary in $builddir/src\n";  }
+if (! -x "$builddir/hmmbuild") { die "FAIL: didn't find hmmconvert binary in $builddir\n";  }
 
 # Create the test file
 open(MSA, ">$tmppfx.sto") || die "FAIL: couldn't create $tmppfx.sto"; 
@@ -31,7 +31,7 @@
 EOF
 close MSA;
 
-$output = `$builddir/src/hmmbuild -O $tmppfx.sto2 --hand $tmppfx.hmm $tmppfx.sto`;
+$output = `$builddir/hmmbuild -O $tmppfx.sto2 --hand $tmppfx.hmm $tmppfx.sto`;
 if ($? != 0) { die "FAIL: hmmbuild failed unexpectedly\n"; }
 
 $output = `grep "^seqx" $tmppfx.sto2`;
