--- hmmer-3.1b2/testsuite/i18-nhmmer-generic.pl	2015-02-26 04:46:05.000000000 -0800
+++ patch-files/i18-nhmmer-generic.pl	2016-05-05 22:37:43.673369143 -0700
@@ -35,13 +35,13 @@
 @eslprogs =  ("esl-shuffle");
 
 # Verify that we have all the executables and datafiles we need for the test.
-foreach $h3prog  (@h3progs)  { if (! -x "$builddir/src/$h3prog")              { die "FAIL: didn't find $h3prog executable in $builddir/src\n";              } }
-foreach $eslprog (@eslprogs) { if (! -x "$builddir/easel/miniapps/$eslprog")  { die "FAIL: didn't find $eslprog executable in $builddir/easel/miniapps\n";  } }
+foreach $h3prog  (@h3progs)  { if (! -x "$builddir/$h3prog")              { die "FAIL: didn't find $h3prog executable in $builddir/\n";              } }
+foreach $eslprog (@eslprogs) { if (! -x "$builddir/$eslprog")  { die "FAIL: didn't find $eslprog executable in $builddir\n";  } }
 
 if (! -r "$srcdir/testsuite/$alignment")  { die "FAIL: can't read msa $alignment in $srcdir/testsuite\n"; }
 
 # Create the test hmm
-$cmd = "$builddir/src/hmmbuild $tmppfx.hmm $srcdir/testsuite/$alignment";
+$cmd = "$builddir/hmmbuild $tmppfx.hmm $srcdir/testsuite/$alignment";
 $output = do_cmd($cmd);
 if ($? != 0) { die "FAIL: hmmbuild failed unexpectedly\n"; } 
 if ($output !~ /1     3box                    22    22    20    75    22.00  1.415/) {
@@ -55,9 +55,9 @@
 
 # Create a roughly 4.5MB database against which to search
 $database   = "$tmppfx.fa";
-do_cmd ( "$builddir/easel/miniapps/esl-shuffle --seed 33 --dna -G -N 1 -L 4500000 -o $tmppfx.A" );
-do_cmd ( "$builddir/src/hmmemit -N 2 --seed 4 $tmppfx.hmm >  $tmppfx.B " );
-do_cmd ( "$builddir/src/hmmemit -N 1 --seed 3 $tmppfx.hmm >> $tmppfx.B" ); 
+do_cmd ( "$builddir/esl-shuffle --seed 33 --dna -G -N 1 -L 4500000 -o $tmppfx.A" );
+do_cmd ( "$builddir/hmmemit -N 2 --seed 4 $tmppfx.hmm >  $tmppfx.B " );
+do_cmd ( "$builddir/hmmemit -N 1 --seed 3 $tmppfx.hmm >> $tmppfx.B" ); 
 do_cmd ( "head -n 33000 $tmppfx.A > $database" );
 do_cmd ( "head -n 2 $tmppfx.B | tail -n 1 >> $database" );
 do_cmd ( "tail -n +33001 $tmppfx.A | head -n 22000 >> $database");
@@ -66,7 +66,7 @@
 do_cmd ( "tail -n 1 $tmppfx.B >> $database" );
 
 # perform nhmmer search
-$cmd = "$builddir/src/nhmmer --tformat fasta $tmppfx.hmm $database";
+$cmd = "$builddir/nhmmer --tformat fasta $tmppfx.hmm $database";
 $output = do_cmd($cmd);
 
 if ($? != 0) { die "FAIL: nhmmer failed unexpectedly\n"; }
@@ -92,7 +92,7 @@
     die "FAIL: nhmmer failed search test 2\n";
 }
 
-$cmd = "$builddir/src/nhmmer --tformat fasta --toponly $tmppfx.hmm $database";
+$cmd = "$builddir/nhmmer --tformat fasta --toponly $tmppfx.hmm $database";
 $output = do_cmd($cmd);
 if ($? != 0) { die "FAIL: nhmmer failed unexpectedly\n"; }
 $expect = 
