--- hmmer-3.2.1/testsuite/i18-nhmmer-generic.pl	2018-06-13 07:39:03.000000000 -0700
+++ patch-files/i18-nhmmer-generic.pl	2018-10-24 17:04:13.938512719 -0700
@@ -33,13 +33,13 @@
 @eslprogs =  ("esl-shuffle");
 
 # Verify that we have all the executables and datafiles we need for the test.
-foreach $h3prog  (@h3progs)  { if (! -x "$builddir/src/$h3prog")              { die "FAIL: didn't find $h3prog executable in $builddir/src\n";              } }
-foreach $eslprog (@eslprogs) { if (! -x "$builddir/easel/miniapps/$eslprog")  { die "FAIL: didn't find $eslprog executable in $builddir/easel/miniapps\n";  } }
+foreach $h3prog  (@h3progs)  { if (! -x "../bin/$h3prog")              { die "FAIL: didn't find $h3prog executable in $builddir\n";              } }
+foreach $eslprog (@eslprogs) { if (! -x "../bin/$eslprog")  { die "FAIL: didn't find $eslprog executable in $builddir/easel/miniapps\n";  } }
 
-if (! -r "$srcdir/testsuite/$alignment")  { die "FAIL: can't read msa $alignment in $srcdir/testsuite\n"; }
+if (! -r "$alignment")  { die "FAIL: can't read msa $alignment in $srcdir/testsuite\n"; }
 
 # Create the test hmm
-$cmd = "$builddir/src/hmmbuild $tmppfx.hmm $srcdir/testsuite/$alignment";
+$cmd = "hmmbuild $tmppfx.hmm $alignment";
 $output = do_cmd($cmd);
 if ($? != 0) { die "FAIL: hmmbuild failed unexpectedly\n"; } 
 if ($output !~ /1     3box                    22    22    20    75    22.00  1.415/) {
@@ -53,9 +53,9 @@
 
 # Create a roughly 4.5MB database against which to search
 $database   = "$tmppfx.fa";
-do_cmd ( "$builddir/easel/miniapps/esl-shuffle --seed 33 --dna -G -N 1 -L 4500000 -o $tmppfx.A" );
-do_cmd ( "$builddir/src/hmmemit -N 2 --seed 4 $tmppfx.hmm >  $tmppfx.B " );
-do_cmd ( "$builddir/src/hmmemit -N 1 --seed 3 $tmppfx.hmm >> $tmppfx.B" ); 
+do_cmd ( "esl-shuffle --seed 33 --dna -G -N 1 -L 4500000 -o $tmppfx.A" );
+do_cmd ( "hmmemit -N 2 --seed 4 $tmppfx.hmm >  $tmppfx.B " );
+do_cmd ( "hmmemit -N 1 --seed 3 $tmppfx.hmm >> $tmppfx.B" ); 
 do_cmd ( "head -n 33000 $tmppfx.A > $database" );
 do_cmd ( "head -n 2 $tmppfx.B | tail -n 1 >> $database" );
 do_cmd ( "tail -n +33001 $tmppfx.A | head -n 22000 >> $database");
@@ -64,7 +64,7 @@
 do_cmd ( "tail -n 1 $tmppfx.B >> $database" );
 
 # perform nhmmer search
-$cmd = "$builddir/src/nhmmer --tformat fasta $tmppfx.hmm $database";
+$cmd = "nhmmer --tformat fasta $tmppfx.hmm $database";
 $output = do_cmd($cmd);
 
 if ($? != 0) { die "FAIL: nhmmer failed unexpectedly\n"; }
@@ -89,7 +89,7 @@
     die "FAIL: nhmmer failed search test 2\n";
 }
 
-$cmd = "$builddir/src/nhmmer --tformat fasta --watson $tmppfx.hmm $database";
+$cmd = "nhmmer --tformat fasta --watson $tmppfx.hmm $database";
 $output = do_cmd($cmd);
 if ($? != 0) { die "FAIL: nhmmer failed unexpectedly\n"; }
 $expect = 
