--- hmmer-3.1b2/testsuite/i19-hmmpgmd-ga.pl	2015-02-26 04:46:05.000000000 -0800
+++ patch-files/i19-hmmpgmd-ga.pl	2016-05-05 22:37:43.673369143 -0700
@@ -27,14 +27,14 @@
 
 # Verify that we have all the executables and datafiles we need for the test.
 @h3progs = ("hmmpgmd", "hmmc2", "hmmpress");
-foreach $h3prog  (@h3progs) { if (! -x "$builddir/src/$h3prog")             { die "FAIL: didn't find $h3prog executable in $builddir/src\n";              } }
+foreach $h3prog  (@h3progs) { if (! -x "$builddir/$h3prog")             { die "FAIL: didn't find $h3prog executable in $builddir\n";              } }
 
 # Verify that threads are enabled in p7_config.h
 # by looking for "#define HMMER_THREADS"
 # if not found, threads are not enabled and this
 # test would fail, but we return ok status because
 # we don't want the full testsuite to fail.
-$have_threads = `cat $builddir/src/p7_config.h | grep "^#define HMMER_THREADS"`;
+$have_threads = `cat $builddir/p7_config.h | grep "^#define HMMER_THREADS"`;
 if($have_threads eq "") { 
     printf("HMMER_THREADS not defined in p7_config.h\n"); 
     exit 0;
@@ -49,20 +49,20 @@
 # create the files needed for the test
 &create_test_hmmdb("$tmppfx.hmm");
 &create_test_script("$tmppfx.in");
-`$builddir/src/hmmpress -f $tmppfx.hmm`;  if ($?) { die "FAIL: hmmpress"; }
+`$builddir/hmmpress -f $tmppfx.hmm`;  if ($?) { die "FAIL: hmmpress"; }
 
 $daemon_active = 0;
 # start the hmmpgmd master and 1 worker with 1 core
-system("$builddir/src/hmmpgmd --master --wport $wport --cport $cport --hmmdb $tmppfx.hmm --pid $tmppfx.pid  > /dev/null 2>&1 &"); 
+system("$builddir/hmmpgmd --master --wport $wport --cport $cport --hmmdb $tmppfx.hmm --pid $tmppfx.pid  > /dev/null 2>&1 &"); 
 if ($?) { die "FAIL: hmmpgmd master failed to start";  }
 $daemon_active = 1;
 sleep 2;  
-system("$builddir/src/hmmpgmd --worker 127.0.0.1 --wport $wport --cpu 1   > /dev/null 2>&1 &"); 
+system("$builddir/hmmpgmd --worker 127.0.0.1 --wport $wport --cpu 1   > /dev/null 2>&1 &"); 
 if ($?) { die "FAIL: hmmpgmd worker failed to start";  }
 sleep 2;
 
 # Run the test script
-@output = `cat $tmppfx.in | $builddir/src/hmmc2 -i $host -p $cport -S 2>&1`;
+@output = `cat $tmppfx.in | $builddir/hmmc2 -i $host -p $cport -S 2>&1`;
 # Currently, hmmc2 returns nonzero exit code even upon clean !shutdown command... don't check $?
 $daemon_active = 0;
 
@@ -111,7 +111,7 @@
 
         # the old way
         #&create_kill_script("$tmppfx.in");
-        #`cat $tmppfx.in | $builddir/src/hmmc2 -i $host -p $cport -S 2>&1`;
+        #`cat $tmppfx.in | $builddir/hmmc2 -i $host -p $cport -S 2>&1`;
     }
     unlink <$tmppfx.hmm*>;
     unlink "$tmppfx.in";
