--- hmmer-3.2.1/testsuite/i20-fmindex-core.pl	2018-06-13 07:39:03.000000000 -0700
+++ patch-files/i20-fmindex-core.pl	2018-10-23 16:16:53.703813874 -0700
@@ -23,11 +23,11 @@
 @h3progs =  ( "makenhmmerdb", "exactmatch");
 
 # Verify that we have all the executables and datafiles we need for the test.
-foreach $h3prog  (@h3progs)  { if (! -x "$builddir/src/$h3prog")              { die "FAIL: didn't find $h3prog executable in $builddir/src\n";              } }
+foreach $h3prog  (@h3progs)  { if (! -x "$builddir/$h3prog")              { die "FAIL: didn't find $h3prog executable in $builddir\n";              } }
 
 &create_db_file("$tmppfx.fa");
 
-$cmd = "$builddir/src/makenhmmerdb $tmppfx.fa $tmppfx.fm 2>&1";
+$cmd = "$builddir/makenhmmerdb $tmppfx.fa $tmppfx.fm 2>&1";
 $output = do_cmd($cmd);
 if ($? != 0) { die "FAIL: makenhmmerdb failed unexpectedly\n"; } 
 if ($output !~ /# alphabet     :                           dna/ ||
@@ -40,7 +40,7 @@
 &create_search_file("$tmppfx.test");
 
 # Search for hits
-$cmd = "$builddir/src/exactmatch --out=- $tmppfx.test $tmppfx.fm "; #2>&1";
+$cmd = "$builddir/exactmatch --out=- $tmppfx.test $tmppfx.fm "; #2>&1";
 $output = do_cmd($cmd);
 if ($? != 0) { die "FAIL: exactmatch failed unexpectedly\n"; }
 $expect = &get_expected();              
