--- hmmer-3.2.1/testsuite/i21-rewind.pl	2018-06-13 07:39:03.000000000 -0700
+++ patch-files/i21-rewind.pl	2018-10-24 18:39:52.336855269 -0700
@@ -24,10 +24,10 @@
 
 # Verify that we have all the executables we need for the test.
 @h3progs =  ( "hmmsearch", "phmmer", "nhmmer");
-foreach $h3prog  (@h3progs)  { if (! -x "$builddir/src/$h3prog")          { die "FAIL: didn't find $h3prog executable in $builddir/src\n";              } }
+foreach $h3prog  (@h3progs)  { if (! -x "../bin/$h3prog")          { die "FAIL: didn't find $h3prog executable in $builddir/src\n";              } }
 
 @easel_progs =  ( "esl-reformat");
-foreach $easel_prog  (@easel_progs)  { if (! -x "$builddir/easel/miniapps/$easel_progs")   { die "FAIL: didn't find $easel_prog executable in $builddir/easel/miniapps\n";              } }
+foreach $easel_prog  (@easel_progs)  { if (! -x "../bin/$easel_progs")   { die "FAIL: didn't find $easel_prog executable in $builddir/easel/miniapps\n";              } }
 
 
 @formats = ("ncbi" , "fasta",  "afa", "stockholm");
@@ -36,14 +36,14 @@
 
 # Test hmmsearch.  Make a query with two copies of the hmm. 
 # Should get the same number of hits with both searches
-$cmd = "cat $srcdir/testsuite/20aa.hmm $srcdir/testsuite/20aa.hmm > $tmppfx.hmm";
+$cmd = "cat 20aa.hmm 20aa.hmm > $tmppfx.hmm";
 do_cmd($cmd);
 
 for $i (0..$#formats) {
    $fmt = $formats[$i];
    $ext = $exts[$i];
 
-   $cmd = "$builddir/src/hmmsearch --tformat $fmt $tmppfx.hmm $srcdir/testsuite/20aa-alitest$ext 2>&1";
+   $cmd = "hmmsearch --tformat $fmt $tmppfx.hmm 20aa-alitest$ext 2>&1";
    $output = do_cmd($cmd);
 
    my ($first)  = ( $output =~ /Domain search space  \(domZ\):\s+(\d+)/g);
@@ -58,7 +58,7 @@
 # Test phmmer.  Make a query with two copies of a sequence. 
 # Should get the same number of hits with both searches
 unlink ("$tmppfx.fa");
-$cmd = "$builddir/src/hmmemit --seed 10 $srcdir/testsuite/20aa.hmm >> $tmppfx.fa";
+$cmd = "hmmemit --seed 10 20aa.hmm >> $tmppfx.fa";
 do_cmd($cmd);
 do_cmd($cmd);  # yes, twice
 
@@ -66,7 +66,7 @@
    $fmt = $formats[$i];
    $ext = $exts[$i];
 
-   $cmd = "$builddir/src/phmmer --tformat $fmt $tmppfx.fa $srcdir/testsuite/20aa-alitest$ext 2>&1";
+   $cmd = "phmmer --tformat $fmt $tmppfx.fa 20aa-alitest$ext 2>&1";
    $output = do_cmd($cmd);
 
    my ($first)  = ( $output =~ /Domain search space  \(domZ\):\s+(\d+)/g);
@@ -80,7 +80,7 @@
 
 # Test nhmmer.  Make a query with two copies of an hmm. 
 # Should get the same number of hits with both searches
-$cmd = "cat $srcdir/testsuite/3box.hmm $srcdir/testsuite/3box.hmm > $tmppfx.hmm";
+$cmd = "cat 3box.hmm 3box.hmm > $tmppfx.hmm";
 do_cmd($cmd);
 
 # the 3box-alitest.fa test was created with:
@@ -100,7 +100,7 @@
    $fmt = $formats[$i];
    $ext = $exts[$i];
 
-   $cmd = "$builddir/src/nhmmer --tformat $fmt $tmppfx.hmm $srcdir/testsuite/3box-alitest$ext 2>&1";
+   $cmd = "nhmmer --tformat $fmt $tmppfx.hmm 3box-alitest$ext 2>&1";
    $output = do_cmd($cmd);
 
    my ($first)  = ( $output =~ /Total number of hits:\s+(\d+)/g);
