--- hmmer-3.2.1/testsuite/i4-zerolength-seqs.sh	2018-06-13 07:39:03.000000000 -0700
+++ patch-files/i4-zerolength-seqs.sh	2018-10-24 17:35:30.813219115 -0700
@@ -24,10 +24,10 @@
 hmmfile=$3;
 tmppfx=$4;
 
-hmmscan=$builddir/src/hmmscan;     if test ! -x $hmmscan;   then echo "FAIL: $hmmscan not executable";   exit 1; fi
-hmmsearch=$builddir/src/hmmsearch; if test ! -x $hmmsearch; then echo "FAIL: $hmmsearch not executable"; exit 1; fi
-phmmer=$builddir/src/phmmer;       if test ! -x $phmmer;    then echo "FAIL: $phmmer not executable";    exit 1; fi
-jackhmmer=$builddir/src/jackhmmer; if test ! -x $jackhmmer; then echo "FAIL: $jackhmmer not executable"; exit 1; fi
+hmmscan=hmmscan;     if test ! -x ../bin/$hmmscan;   then echo "FAIL: $hmmscan not executable";   exit 1; fi
+hmmsearch=hmmsearch; if test ! -x ../bin/$hmmsearch; then echo "FAIL: $hmmsearch not executable"; exit 1; fi
+phmmer=phmmer;       if test ! -x ../bin/$phmmer;    then echo "FAIL: $phmmer not executable";    exit 1; fi
+jackhmmer=jackhmmer; if test ! -x ../bin/$jackhmmer; then echo "FAIL: $jackhmmer not executable"; exit 1; fi
                                    if test ! -r $hmmfile;   then echo "FAIL: $hmmfile not readable";     exit 1; fi
 
 fafile=$tmppfx.fa
