--- hmmer-3.2.1/testsuite/i7-hmmbuild-fragments.pl	2018-06-13 07:39:03.000000000 -0700
+++ patch-files/i7-hmmbuild-fragments.pl	2018-10-23 16:17:57.896293037 -0700
@@ -4,7 +4,7 @@
 # of lots of sequence fragments.
 #
 # Usage:    ./i7-hmmbuild-fragments.pl <hmmbuild binary> <tmpfile prefix>
-# Example:  ./i7-hmmbuild-fragments.pl ../src/hmmbuild   tmpfoo
+# Example:  ./i7-hmmbuild-fragments.pl ../hmmbuild   tmpfoo
 # 
 # SRE, Tue Jun 16 13:37:05 2009
 
