--- BS_Seeker.py	2010-03-26 10:55:43.000000000 -0700
+++ patch-files/BS_Seeker.py	2014-01-09 17:09:00.000000000 -0800
@@ -102,7 +102,7 @@
 parser.add_option("-a", "--adapter", dest="adapterfilename",help="Input your adapter file", metavar="FILE")
 
 
-parser.set_defaults(bowtiepath="~/bowtie-0.10.0/")
+parser.set_defaults(bowtiepath="BOWTIEPATH")
 parser.add_option("-p", "--path", dest="bowtiepath",help="Path to Bowtie [~/bowtie-0.10.0/]", metavar="PATH")
 
 parser.set_defaults(dbpath="./reference_genome/")
