--- Preprocessing_genome.py	2010-03-26 08:52:18.000000000 -0700
+++ patch-files/Preprocessing_genome.py	2014-01-09 17:09:10.000000000 -0800
@@ -14,7 +14,7 @@
 parser.set_defaults(taginfo="Y")
 parser.add_option("-t", "--tag", dest="taginfo",help="reads with tags? Y/N  [Y]", metavar="TAG")
 
-parser.set_defaults(bowtiepath="~/bowtie-0.10.0/")
+parser.set_defaults(bowtiepath="BOWTIEPATH")
 parser.add_option("-p", "--path", dest="bowtiepath",help="Path to Bowtie [~/bowtie-0.10.0/]", metavar="PATH")
 
 (options, args) = parser.parse_args()
