--- vcftools-vcftools-c7a7337/configure.ac	2015-09-15 11:04:29.000000000 -0700
+++ patch-files/configure.ac	2015-09-27 21:49:45.472666143 -0700
@@ -39,7 +39,8 @@
 AC_SUBST([PMDIR])
 
 # Checks for libraries.
-PKG_CHECK_MODULES(ZLIB, zlib)
+#PKG_CHECK_MODULES(ZLIB, zlib)
+AC_CHECK_LIB(libz, libz)
 
 # Checks for header files.
 AC_CHECK_HEADERS([arpa/inet.h fcntl.h limits.h netdb.h stdint.h stdlib.h string.h sys/socket.h unistd.h])
