--- trinityrnaseq_r20140717/trinity-plugins/Makefile	2014-07-16 18:17:43.000000000 -0700
+++ patch-files/Makefile	2015-03-25 14:09:16.643776655 -0700
@@ -16,7 +16,14 @@
 
 rsem:
 	tar -zxvf ${RSEM_CODE}.tar.gz && ln -sf ${RSEM_CODE} tmp.rsem
-	cd ./tmp.rsem && $(MAKE)
+	( \
+            cd ./tmp.rsem; \
+            cd sam; \
+            cat Makefile|sed 's/-lcurses/-lcurses -ltinfo/' >Makefile.tmp; \
+            mv Makefile.tmp Makefile; \
+            cd .. ; \
+            $(MAKE) \
+        )
 	mv tmp.rsem rsem
 
 transdecoder:
