--- trinityrnaseq_r20131110/sample_data/test_Trinity_Assembly/runMe.sh	2013-11-10 14:37:57.000000000 -0800
+++ patch-files/runMe.sh	2014-02-25 21:08:02.000000000 -0800
@@ -16,7 +16,7 @@
 #######################################################
 
 ## use jellyfish
-../../Trinity.pl --seqType fq --JM 2G --left reads.left.fq --right reads.right.fq --SS_lib_type RF --CPU 4 # --bowtie_comp
+ROLLPREFIX/Trinity.pl --seqType fq --JM 2G --left reads.left.fq --right reads.right.fq --SS_lib_type RF --CPU 4 # --bowtie_comp
 
 ##### Done Running Trinity #####
 
@@ -34,7 +34,7 @@
 
 sleep 2
 
-../../util/RSEM_util/run_RSEM_align_n_estimate.pl  --transcripts trinity_out_dir/Trinity.fasta --seqType fq --left reads.left.fq --right reads.right.fq --SS_lib_type RF 
+ROLLPREFIX/util/RSEM_util/run_RSEM_align_n_estimate.pl  --transcripts trinity_out_dir/Trinity.fasta --seqType fq --left reads.left.fq --right reads.right.fq --SS_lib_type RF 
 
 ###### Done running RSEM ########
 
