--- trinityrnaseq-2.0.6/sample_data/test_Trinity_Assembly/runMe.sh	2015-03-02 10:08:04.000000000 -0800
+++ patch-files/runMe.sh	2015-09-27 18:21:10.284967876 -0700
@@ -23,7 +23,7 @@
 ##  Run Trinity to Generate Transcriptome Assemblies ##
 #######################################################
 
-../../Trinity --seqType fq --max_memory 2G --left reads.left.fq.gz,reads2.left.fq.gz --right reads.right.fq.gz,reads2.right.fq.gz --SS_lib_type RF --CPU 4
+ROLLPREFIX/Trinity --seqType fq --max_memory 2G --left reads.left.fq.gz,reads2.left.fq.gz --right reads.right.fq.gz,reads2.right.fq.gz --SS_lib_type RF --CPU 4
 
 ##### Done Running Trinity #####
 
@@ -42,10 +42,10 @@
 sleep 2
 
 # first try RSEM
-../../util/align_and_estimate_abundance.pl --transcripts trinity_out_dir/Trinity.fasta --seqType fq --left reads.left.fq --right reads.right.fq --SS_lib_type RF --est_method RSEM --aln_method bowtie --trinity_mode --prep_reference
+ROLLPREFIX/util/align_and_estimate_abundance.pl --transcripts trinity_out_dir/Trinity.fasta --seqType fq --left reads.left.fq --right reads.right.fq --SS_lib_type RF --est_method RSEM --aln_method bowtie --trinity_mode --prep_reference
 
 # try eXpress
-../../util/align_and_estimate_abundance.pl --transcripts trinity_out_dir/Trinity.fasta --seqType fq --left reads.left.fq --right reads.right.fq --SS_lib_type RF --est_method eXpress --aln_method bowtie2 --trinity_mode --prep_reference
+ROLLPREFIX/util/align_and_estimate_abundance.pl --transcripts trinity_out_dir/Trinity.fasta --seqType fq --left reads.left.fq --right reads.right.fq --SS_lib_type RF --est_method eXpress --aln_method bowtie2 --trinity_mode --prep_reference
 
 
 ###### Done running RSEM ########
