--- trinityrnaseq_r20131110/Trinity.pl	2013-11-10 14:42:41.000000000 -0800
+++ patch-files/Trinity.pl	2014-02-25 17:36:43.000000000 -0800
@@ -719,7 +719,7 @@
 my $PAIRED_MODE = ( (@left_files && @right_files)  || $run_as_paired_flag) ? 1:0;
 if ($PAIRED_MODE && (!$NO_RUN_CHRYSALIS_FLAG)  && (!$NO_BOWTIE)) {
     ## be sure we can find 'bowtie', since we use it as part of the iworm pair scaffolding step
-    my $bowtie_path = `which bowtie`;
+    my $bowtie_path = `export PATH=ROLLBIOTOOLS/bowtie/bin:\$PATH;which bowtie`;
     if ($bowtie_path =~ /\w/) {
         print "Paired mode requires bowtie. Found bowtie at: $bowtie_path\n";
     }
@@ -727,7 +727,7 @@
         die "Error, cannot find path to bowtie, which is now needed as part of Chrysalis' read scaffolding step";
     }
     
-    my $samtools_path = `which samtools`;
+    my $samtools_path = `export PATH=ROLLBIOTOOLS/samtools/bin:\$PATH;which samtools`;
     if ($samtools_path =~ /\w/) {
         print "Found samtools at: $samtools_path\n";
     }
@@ -943,9 +943,9 @@
         $MKDIR_OUTDIR_FLAG = 1;
     }
 
-    unless (-d $chrysalis_output_dir) {
-        &process_cmd("mkdir -p $chrysalis_output_dir"); # note, won't be auto-cleaned up if not in the trinity_out_dir/
-    }
+#   unless (-d $chrysalis_output_dir) {
+#       &process_cmd("mkdir -p $chrysalis_output_dir"); # note, won't be auto-cleaned up if not in the trinity_out_dir/
+#   }
     
     chdir ($output_directory) or die "Error, cannot cd to $output_directory";
     
@@ -1304,7 +1304,7 @@
         
         eval {
             
-            &process_cmd($cmd);
+            &process_cmd("export PATH=ROLLBIOTOOLS/bowtie/bin:\$PATH; $cmd");
             
         };
         
