--- SPAdes-3.13.0/spades_compile.sh	2018-10-12 03:31:36.000000000 -0700
+++ patch-files/spades_compile.sh	2018-10-29 16:28:41.309668922 -0700
@@ -19,7 +19,7 @@
 mkdir -p "$BASEDIR/$BUILD_DIR"
 set -e
 cd "$BASEDIR/$BUILD_DIR"
-cmake -G "Unix Makefiles" -DCMAKE_INSTALL_PREFIX="$PREFIX" $* "$BASEDIR/src"
+cmake -G "Unix Makefiles" -DCMAKE_C_COMPILER=ROLLPATHCC -DBoost_NO_BOOST_CMAKE=ON -DCMAKE_INSTALL_PREFIX=$PREFIX $BASEDIR/src $*
 make -j 8
 make install
 cd $PREFIX
